go/oprf/groups/ecgroup/h2c.go

```diff
@@ -143,11 +159,29 @@ func (params h2cParams) hashToCurve(alpha []byte) (Point, error) {
 		return Point{}, oerr.ErrIncompatibleGroupParams
 	}
 
-	// construct the output point R
-	P, err := R.clearCofactor(params.hEff)
+	geR, err := Q0.Add(Q1)
+	if err != nil {
+		return Point{}, err
+	}
+
+	R, err = castToPoint(geR)
 	if err != nil {
 		return Point{}, err
 	}
+
+	P := Point{}.New(params.gc).(Point)
+
+	// construct the output point R
+	if params.name == "SSWU-R0" {
+		P, err = R.clearCofactor(params.hEff)
```

Comment:
This is interesting (thanks to your comment for context!). It's also a redundant check for P-384 and P-521 as they have cofactor 1. It might be better to have an extra boolean included in the `h2cParams` struct that specifies whether they need cofactors clearing, or not. Then we could only perform this check if the boolean is true.

Reply:
Yeah! I agree on this.. maybe it should be deleted completely?